spark3/src/main/java/org/apache/iceberg/actions/Spark3SnapshotAction.java

```diff
@@ -135,10 +140,23 @@ protected TableCatalog checkSourceCatalog(CatalogPlugin catalog) {
 
   @Override
   public SnapshotAction withLocation(String location) {
-    Preconditions.checkArgument(!sourceTableLocation().equals(location),
```

Comment:
`sourceTableLocation` is converted from URI, while `location` comes from SQL command. We should qualify the `location` before check equals.

Reply:
Good catch!

Reply:
We should probably also check if two locations overlap. We will corrupt the original table if the source location is within the original location.

Reply:
Does it mean we need to normalize and resolve the locations?